### ISSUE TITLE
Monitor disabling problem

### ISSUE THREAD

**Issue by dusanx** [hyprwm/Hyprland#8790]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

Yes

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland 0.45.2 built from branch  at commit 12f9a0d0b93f691d4d9923716557154d74777b0a  ([gha] Nix: update inputs).
Date: Tue Nov 19 21:47:18 2024
Tag: v0.45.2, commits: 5451
built against aquamarine 0.5.0


flags set:
debug


System Information:
System name: Linux
Node name: ryzen
Release: 6.12.4-artix1-1
Version: #1 SMP PREEMPT_DYNAMIC Mon, 09 Dec 2024 17:50:14 +0000


GPU information:
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation TU117M [GeForce GTX 1650 Mobile / Max-Q] [10de:1f99] (rev a1) (prog-if 00 [VGA controller])
05:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Cezanne [Radeon Vega Series / Radeon Vega Mobile Series] [1002:1638] (rev c6) (prog-if 00 [VGA controller])


os-release: NAME="Artix Linux"
PRETTY_NAME="Artix Linux"
ID=artix
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://artixlinux.org/"
DOCUMENTATION_URL="https://wiki.artixlinux.org/"
SUPPORT_URL="https://forum.artixlinux.org/"
BUG_REPORT_URL="https://bugs.artixlinux.org/"
PRIVACY_POLICY_URL="https://terms.artixlinux.org/docs/privacy-policy/"
LOGO=artixlinux-logo


plugins:

======Config-Start======


Config File: /home/dusanx/.config/hypr/hyprland.conf: Read Succeeded

#
#AUTOGENERATED HYPR CONFIG.
#PLEASE USE THE CONFIG PROVIDED IN THE GIT REPO /examples/hypr.conf AND EDIT IT,
#OR EDIT THIS ONE ACCORDING TO THE WIKI INSTRUCTIONS.
#

#
# Please note not all available settings / options are set here.
# For a full list, see the wiki
#

#autogenerated = 1 # remove this line to remove the warning

# See https://wiki.hyprland.org/Configuring/Monitors/
#monitor=,preferred,auto,auto
#monitor=name,resolution,position,scale
#monitor=DP-1,1920x1080@144,0x0,1
#####monitor=HDMI-A-1,1920x1080,auto,1.0
#########################monitor=HDMI-A-1,1920x1080@165,auto,1.0
#monitor=eDP-1,preferred,auto,auto,mirror,HDMI-A-1

# Sve ide na veliki monitor
monitor=DP-1,1920x1080@165,0x0,1
#monitor=eDP-1,disable

# Sve ide na laptop monitor
# monitor=DP-1,disable
# monitor=eDP-1,1920x1080,auto,1.0


# See https://wiki.hyprland.org/Configuring/Keywords/ for more

# Execute your favorite apps at launch
# exec-once = waybar & hyprpaper & firefox
exec-once = waybar
exec-once = swaync
# exec-once = emacs --daemon
#exec-once = swaybg -m fill -i ~/media/wallpaper/haskell-lambda-temnyy-fon.jpeg
#exec-once = swaybg -m fill -i ~/media/wallpaper/old-black-background-grunge-texture-dark-wallpaper-blackboard-chalkboard-room-wall.jpg
#exec-once = swaybg -m fill -i ~/media/wallpaper/haskell.png
exec-once = swaybg -m fill -i ~/media/wallpaper/applikely-wallpaper.png
#exec-once = swaybg -m fill -i ~/media/wallpaper/wallpaperbetter.com_1920x1080.jpg
#exec-once = swaybg -m fill -i ~/media/wallpaper/novi/digital-art-blue-leaves-pattern-wallpaper-29b0f8ed811a8d3b1687089f90d1367d.jpg
#exec-once = swaybg -m fill -i ~/media/wallpaper/novi/pattern-dark-texture-black-wallpaper-d9d0082d611addcb3637a86f40e1d61d.jpg
#exec-once = swaybg -m fill -i ~/media/wallpaper/minimalist-black-wallpaper.png
#exec-once = swaybg -m fill -i ~/media/wallpaper/wallpaperbetter.com_1920x1080.jpg
exec-once = ~/bin/startpipewire.sh
#exec-once = sleep 3 && conky -c ~/bin/conky/conkyrc-cal
#exec-once = sleep 3 && conky -c ~/bin/conky/conkyrc-todo
exec-once = ~/bin/wayland/sleep
exec-once = mako
#exec-once = nwg-dock-hyprland -x
exec-once = nwg-dock-hyprland -x -ico app-launcher -i 40
#exec-once = ~/projects/git/nwg-dock-hyprland/bin/nwg-dock-hyprland -x -ico app-launcher
exec-once = hyprctl keyword monitor "eDP-1, disable"

# Source a file (multi-file configs)
# source = ~/.config/hypr/myColors.conf

# Set programs that you use
$terminal = kitty
$fileManager = dolphin
#$menu = wofi --show drun
$menu = fuzzel

# Some default env vars.
#env = XCURSOR_THEME,Bibata Red
#env = XCURSOR_SIZE,24
#env = WLR_DRM_DEVICES,/dev/dri/card2:/dev/dri/card1
#env = AQ_DRM_DEVICES,/dev/dri/card2:/dev/dri/card1
#env = AQ_DRM_DEVICES,/dev/dri/card1:/dev/dri/card2
#env = AQ_DRM_DEVICES,/dev/dri/card1

# env = WLR_DRM_DEVICES,$HOME/.config/hypr/card

# For all categories, see https://wiki.hyprland.org/Configuring/Variables/
input {
    kb_layout = us
    kb_variant =
    kb_model =
    kb_options =
    kb_rules =

    follow_mouse = 0

    touchpad {
        natural_scroll = no
    }

    sensitivity = 0 # -1.0 - 1.0, 0 means no modification.

    repeat_rate = 70
    repeat_delay = 200
}

general {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    # cursor_inactive_timeout = 1

    gaps_in = 6
    gaps_out = 12
    #border_size = 2
    border_size = 2
    col.active_border = rgba(33ccffee) rgba(00ff99ee) 45deg
    col.inactive_border = rgba(595959aa)

    layout = master

    # Please see https://wiki.hyprland.org/Configuring/Tearing/ before you turn this on
    allow_tearing = false
}

cursor {
    inactive_timeout = 1
    enable_hyprcursor = true
    no_warps = true
    no_hardware_cursors = true
}

decoration {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    #rounding = 15
    rounding = 5

    blur {
        enabled = true
        size = 3
        passes = 1
    }

    #DEPRECATED? drop_shadow = true
    #DEPRECATED? shadow_range = 4
    #DEPRECATED? shadow_render_power = 3
    #DEPRECATED? col.shadow = rgba(1a1a1aee)

    #shadow_offset = 2 2

    dim_inactive = false
}

animations {
    enabled = yes # bilo yes

    # Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more

    bezier = myBezier, 0.05, 0.9, 0.1, 1.05

    animation = windows, 1, 7, myBezier
    animation = windowsOut, 1, 7, default, popin 80%
    animation = border, 1, 10, default
    animation = borderangle, 1, 8, default
    animation = fade, 1, 7, default
    animation = workspaces, 1, 6, default
}

dwindle {
    # See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    pseudotile = yes # master switch for pseudotiling. Enabling is bound to mainMod + P in the keybinds section below
    preserve_split = yes # you probably want this
}

master {
    # See https://wiki.hyprland.org/Configuring/Master-Layout/ for more
    # FIXME: VIDI OVO ŠTA JE ZAMENA new_is_master = true
    new_on_top = true
    mfact = 0.5
}

gestures {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    workspace_swipe = off
}

misc {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    force_default_wallpaper = 0 # Set to 0 to disable the anime mascot wallpapers
    mouse_move_enables_dpms = true
    key_press_enables_dpms = true
    disable_hyprland_logo = true
}

binds {
    scroll_event_delay = 100
}

# Example per-device config
# See https://wiki.hyprland.org/Configuring/Keywords/#executing for more
#ovodevice:epic-mouse-v1 {
#ovo    sensitivity = -0.5
#ovo}

# Example windowrule v1
# windowrule = float, ^(kitty)$
# Example windowrule v2
# windowrulev2 = float,class:^(kitty)$,title:^(kitty)$
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more
#ovowindowrulev2 = nomaximizerequest, class:.* # You'll probably like this.
#windowrulev2 = tile,class:(Godot)
#windowrulev2 = tile,initialTitle:Godot


# See https://wiki.hyprland.org/Configuring/Keywords/ for more
$mainMod = SUPER

# Example binds, see https://wiki.hyprland.org/Configuring/Binds/ for more
# bind = $mainMod SHIFT, RETURN, exec, $terminal
#bind = $mainMod SHIFT, RETURN, exec, hyprctl dispatch movecursor 500 0 && $terminal
bind = $mainMod SHIFT, RETURN, exec, $terminal
bind = $mainMod SHIFT, C, killactive,
#bind = $mainMod SHIFT CTRL, Q, exit,
bind = $mainMod SHIFT CTRL, Q, exec, ~/bin/killhypr
#bind = $mainMod SHIFT, E, exec, hyprctl dispatch movecursor 500 0 && emacs
bind = $mainMod SHIFT, E, exec, emacs
#bind = $mainMod SHIFT, E, exec, emacsclient -c
bind = $mainMod, T, togglefloating,
bind = $mainMod, G, exec, hyprctl dispatch movecursor 500 0 && $menu
bind = SUPER CTRL ALT, P, exec, dpass
#bind = $mainMod, G, exec, $menu
#???? bind = $mainMod, W, pseudo, # dwindle
# bind = $mainMod, W, togglegroup # tabovi

# dwindle layout
# bind = $mainMod, J, togglesplit, # dwindle
bind = $mainMod, N, movefocus, l # Move focus with mainMod + arrow keys
bind = $mainMod, O, movefocus, r
bind = $mainMod, I, movefocus, u
bind = $mainMod, E, movefocus, d

# master layout
bind = $mainMod, SPACE, layoutmsg, orientationcycle left top # master
bind = $mainMod SHIFT, N, layoutmsg, swapprev # master
bind = $mainMod SHIFT, O, layoutmsg, swapnext # master
# bind = $mainMod, N, layoutmsg, cycleprev
# bind = $mainMod, O, layoutmsg, cyclenext
bind = SHIFT, RETURN, layoutmsg, swapwithmaster auto

# Move window
#bind = $mainMod SHIFT, N, movewindow, l
#bind = $mainMod SHIFT, O, movewindow, r
#bind = $mainMod SHIFT, I, movewindow, u
#bind = $mainMod SHIFT, E, movewindow, d

# Floating
bind = $mainMod, w, exec, hyprfloating

# Switch workspaces with mainMod + [0-9]
bind = $mainMod, 1, workspace, 1
bind = $mainMod, 2, workspace, 2
bind = $mainMod, 3, workspace, 3
bind = $mainMod, 4, workspace, 4
bind = $mainMod, 5, workspace, 5
bind = $mainMod, 6, workspace, 6
bind = $mainMod, 7, workspace, 7
bind = $mainMod, 8, workspace, 8
bind = $mainMod, 9, workspace, 9
bind = $mainMod, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspacesilent, 1
bind = $mainMod SHIFT, 2, movetoworkspacesilent, 2
bind = $mainMod SHIFT, 3, movetoworkspacesilent, 3
bind = $mainMod SHIFT, 4, movetoworkspacesilent, 4
bind = $mainMod SHIFT, 5, movetoworkspacesilent, 5
bind = $mainMod SHIFT, 6, movetoworkspacesilent, 6
bind = $mainMod SHIFT, 7, movetoworkspacesilent, 7
bind = $mainMod SHIFT, 8, movetoworkspacesilent, 8
bind = $mainMod SHIFT, 9, movetoworkspacesilent, 9
bind = $mainMod SHIFT, 0, movetoworkspacesilent, 10

# Example special workspace (scratchpad)
# bind = $mainMod, S, togglespecialworkspace, magic
# bind = $mainMod SHIFT, S, movetoworkspace, special:magic

# Scroll through existing workspaces with mainMod + scroll
#bind = $mainMod, mouse_down, workspace, e+1
#bind = $mainMod, mouse_up, workspace, e-1

# Fullscreen
bind = $mainMod, F, fullscreen
# bind = $mainMod, F, fullscreen, 1 # maximized
# bind = $mainMod SHIFT, F, fullscreen # pravi full screen

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:272, movewindow
bindm = $mainMod, mouse:273, resizewindow

# bind = , XF86MonBrightnessUp, exec, ~/bin/brightness_control_hdmi up
# bind = , XF86MonBrightnessDown, exec, ~/bin/brightness_control_hdmi down

bind = , XF86AudioPlay, exec, ~/bin/mpctoggle
#bind = , XF86AudioNext, exec, hyprctl dispatch movecursor 500 0 && kitty -e ncmpc
bind = , XF86AudioNext, exec, kitty -e ncmpc

bind =  SUPER CTRL ALT, W, exec, grim -g "$(slurp)"
bind =  SUPER CTRL ALT, D, exec, grim && notify-send -t 100 "Screnshot"

bind = $mainMod, mouse_up, exec, ~/bin/wayland/zoom.py minus
bind = $mainMod, mouse_down, exec, ~/bin/wayland/zoom.py plus

# bind = $mainMod, V, exec, cliphist list | wofi --dmenu | cliphist decode | wl-copy

#bindl=,switch:off:Lid Switch,exec,hyprctl keyword monitor "eDP-1,preferred,auto,auto,mirror,HDMI-A-1" ; notify-send "upalio monitor"
#bindl=,switch:on:Lid Switch,exec,hyprctl keyword monitor "eDP-1, disable" ; notify-send "ugasio monitor"

======Config-End========
```

</details>

### Description

I have two monitors, one internal in laptop (eDP-1), other HP hooked trough HDMI. I only need external monitor, so I want to turn off one on the laptop. I also don't use any greeter, I log into console instead and execute hyprland manually to start wayland.

If I define laptop monitor as disabled
```
monitor=DP-1,1920x1080@165,0x0,1
monitor=eDP-1,disable
```
when hyprland starts it is not active but doesn't turn off and keeps console text visible (whatever was on console before starting hyprland).

However if I
```
hyprctl keyword monitor "eDP-1, disable"
```
then laptop monitor turns off.

or if I add
```
exec-once = hyprctl keyword monitor "eDP-1, disable"
```
it also turns off.

I suspect it is some timing problem where ```monitor=``` doesn't work but other two solutions do.

### How to reproduce

No idea as it might be laptop specific but I know it worked up until recently.

**Comment by ewwwdp** [comment]
Same issue, `hyprctl keyword monitor "eDP-1, disable"` doesn't turn off monitor.
### System info:
```
Hyprland 0.46.2 built from branch  at commit 0bd541f2fd902dbfa04c3ea2ccf679395e316887  (version: bump to 0.46.2).
Date: Thu Dec 19 19:26:47 2024
Tag: v0.46.2, commits: 5566
built against:
 aquamarine 0.5.1
 hyprlang 0.6.0
 hyprutils 0.2.6
 hyprcursor 0.1.10
 hyprgraphics 0.1.1


flags set:
debug


System Information:
System name: Linux
Node name: dopler-fedora
Release: 6.12.5-200.fc41.x86_64
Version: #1 SMP PREEMPT_DYNAMIC Sun Dec 15 16:48:23 UTC 2024


GPU information:
00:02.0 VGA compatible controller [0300]: Intel Corporation Iris Plus Graphics G1 (Ice Lake) [8086:8a56] (rev 07) (prog-if 00 [VGA controller])


os-release: NAME="Fedora Linux"
VERSION="41 (Sway)"
RELEASE_TYPE=stable
ID=fedora
VERSION_ID=41
VERSION_CODENAME=""
PLATFORM_ID="platform:f41"
PRETTY_NAME="Fedora Linux 41 (Sway)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:41"
DEFAULT_HOSTNAME="fedora"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f41/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://gitlab.com/fedora/sigs/sway/SIG/-/issues"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=41
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=41
SUPPORT_END=2025-12-15
VARIANT="Sway"
VARIANT_ID=sway
```

**Comment by ewwwdp** [comment]
Issue still exists after upgrading to aquamarine 0.6.0. @dusanx can u check on your side?

**Comment by youwen5** [comment]
I'm also experiencing this issue, using external monitor with laptop screen set to disable and it stays on after Hyprland starts with the console logs. Tried building Hyprland 0.46.2 against Aquamarine 0.6.0 but doesn't work. Running NixOS unstable.

```
Hyprland 0.46.2 built from branch v0.46.2-b at commit 0bd541f2fd902dbfa04c3ea2ccf679395e316887  (version: bump to 0.46.2).
Date: 2024-12-19
Tag: v0.46.2, commits: 0bd541f2fd902dbfa04c3ea2ccf679395e316887
built against:
 aquamarine 0.6.0
 hyprlang 0.6.0
 hyprutils 0.3.0
 hyprcursor 0.1.11
 hyprgraphics 0.1.1


flags set:
debug
```

**Comment by ewwwdp** [comment]
@youwen5 this issue was fixed for me after upgrading to aquamarine v0.8.0.
